src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -385,11 +387,13 @@ def adjustJSONSpec(self, puWflows, msPileupList, dest=None):
                         # {block1: [rses], block2: [rses], ...}
                         if checkChanges(jsonBlockLoc, msPUBlockLoc):
                             self.logger.info("Found differences between JSON and MSPileup content.")
-                            puNewJsonContent = updateBlockInfo(puJsonContent, msPUBlockLoc, self.logger)
+                            puNewJsonContent = updateBlockInfo(puJsonContent, msPUBlockLoc, self.dbsUrl, self.logger)
 
                             # we should update a tarball only once for each pileup name,
                             # therefore we add new entry to jdict with our pilupe conf file
-                            jdict[puConfJson] = puNewJsonContent
+                            if puNewJsonContent:
+                                # we update json file if we get new pileup content
+                                jdict[puConfJson] = puNewJsonContent
```

Comment:
maybe we should add an `else` construct to log a message indicating that there was a change in the Json file and the MSpilup configuration but the update has fallen, so that during debugging it would become immediately clear where to look for the broken stop (e.g. None value returned  by DBS or Rucio calls etc.... )

Reply:
Todor, good point, I updated PR to add appropriate logger message.